markdown/how.md

```diff
@@ -3,17 +3,68 @@
 
 ## OpenStack
 
+<!-- Note -->
+
 
 ## Magnum
 
+<!-- Note -->
+
+In order to transition to the Docker-based deployment model with Kubernetes, we required a container orchestration service and a container registry offering for storing private container images.
+Within our cloud infrastructure, we had two options available for container orchestration services: Rancher and Magnum. We chose OpenStack Magnum because it provided all the necessary components for a production-ready setup for deploying Tutor.
+
 
 ### Private registry
 
+<!-- Note -->
+
+We needed a container orchestration service that could accommodate custom-built Docker images, and thus we decided to utilize the private Docker registry provided by Magnum. The built-in Docker registry runs locally on each Kubernetes's worker nodes localhost. It does not require authentication, making it easy to publish and pull images without any configuration changes to Tutor.
+
+The Docker registry supported by Magnum utilizes Swift for storage and does not maintain any local state. Therefore, it allows us to launch it as needed in any CI environment,as long as we can provide the necessary Swift credentials.
+
 
 ## Kubernetes
 
+<!-- Note -->
+
+Octavia is the OpenStack load balancing service. We create a Kubernetes cluster with Magnum and use an Octavia Load Balancer to access it from the public network, via the the load balancer’s floating IP.
```

Comment:
It makes no sense to immediately start talking about Octavia when the slide says "Kubernetes". This should probably go to the end of the notes for this slide.